common/exception/TypeDBClientException.java

```diff
@@ -48,7 +53,13 @@ public TypeDBClientException(ErrorMessage error, Object... parameters) {
         this.errorMessage = error;
     }
 
-    public TypeDBClientException(String message, Throwable cause) {
+    public TypeDBClientException(Throwable cause, ErrorMessage error, Object... parameters) {
+        super(error.message(parameters), cause);
+        assert !getMessage().contains("%s");
```

Comment:
I don't think this should be possible since the string formatting should fail if the number of args doesn't match? Otherwise I think this should move into the `error.message()` call since we'd like to encapsulate that behaviour in the creation of the message from the error

Reply:
Ok maybe you can update that one too ;)